TPL/force-app/main/default/classes/HealthcareCostTriggerHandler.cls

```diff
@@ -7,47 +7,159 @@ public with sharing class HealthcareCostTriggerHandler {
-                c = caseMap.get((String)agg.get('case2__c'));
-            }else{
-                c.id = (String)agg.get('case2__c');
-            }
-            if((String)agg.get('DeveloperName') == 'Pharmacare'){
+        Decimal totalCostOverride =0;
+        Decimal totalCost = 0;
+        Map<Id,Case> caseMap = new Map<Id,Case>();
+        Map<Id,Integer> totalCostMap = new Map<Id,Integer>();
+        Map<Id,Map<String,Decimal>> caseIdRecordTypeSumMap = new Map<Id,Map<String,Decimal>>();
+        Map<Id,Map<String,Decimal>> caseIdRecordTypeCostMap = new Map<Id,Map<String,Decimal>>();
+        
+        for(Healthcare_Cost__c hcc : hccSumList){
+            System.debug(hcc);
+            if(hcc.RecordTypeName__c != 'Continuing_Care' && hcc.RecordTypeName__c != 'Future_Care'){
+                if(caseIdRecordTypeSumMap.containsKey(hcc.Case2__c)){
+                    if(hcc.Cost_Include__c && hcc.Cost_Review__c){
+                        Map<String,Decimal> tempMap = caseIdRecordTypeSumMap.get(hcc.Case2__c);
+                        System.debug(tempMap);
+                        if(tempMap.containskey(hcc.RecordTypeName__c)){
+                            totalCostOverride = tempMap.get(hcc.RecordTypeName__c);
+                            totalCostOverride = Integer.valueof(totalCostOverride + hcc.Total_Cost_Override__c);
+                            tempMap.put(hcc.RecordTypeName__c,totalCostOverride);
+                        }else{
+                            tempMap.put(hcc.RecordTypeName__c,hcc.Total_Cost_Override__c);
+                        } 
+                        caseIdRecordTypeSumMap.put(hcc.Case2__c,tempMap);
+                    }
+                    
+                }else{
+                    Map<String,Decimal> tempMap = new Map<String,Decimal>();
+                    if(hcc.Cost_Include__c && hcc.Cost_Review__c){
+                        tempMap.put(hcc.RecordTypeName__c,hcc.Total_Cost_Override__c);
+                        System.debug(hcc.Id+'    ');
+                        caseIdRecordTypeSumMap.put(hcc.Case2__c,tempMap);
+                    }else{
+                        tempMap.put(hcc.RecordTypeName__c,0);
+                        System.debug(hcc.Id+'    ');
+                        caseIdRecordTypeSumMap.put(hcc.Case2__c,tempMap);
+                    }
+                }
                 
-                c.Pharmacare_Total2__c = (Decimal)agg.get('tot');
-               
-            }
-            if((String)agg.get('DeveloperName') == 'Continuing_Care'){
-               
-                c.Continuing_Care_Total2__c = (Decimal)agg.get('costSum');
+            }else{
+                if(caseIdRecordTypeCostMap.containsKey(hcc.Case2__c)){
+                    Map<String,Decimal> tempMap = caseIdRecordTypeCostMap.get(hcc.Case2__c);
+                    if(tempMap.containskey(hcc.RecordTypeName__c)){
+                        System.debug(hcc.RecordTypeName__c);
```

Comment:
If you wish to retain the debug cover them with a boolean constant to deactivate them in the class opening statements.